cmd/srrs/main.go

```diff
@@ -2,109 +2,127 @@ package main
-		},
-	}
+
+	stateEndpoint   = path.Join("api", "v1", "state")
+	turtleEndpoint  = path.Join("api", "v1", "turtles")
+	commandEndpoint = path.Join("api", "v1", "command")
 )
 
 func init() {
-	log.SetFlags(0)
+	log.SetLevel(log.InfoLevel)
+	log.SetFormatter(&log.TextFormatter{
+		ForceColors:      true,
+		DisableTimestamp: true,
+		QuoteEmptyFields: true,
+	})
+	log.SetOutput(os.Stdout)
 }
 
 func main() {
 	flag.Parse()
 
-	http.HandleFunc("/"+defaultEndpoint, func(w http.ResponseWriter, r *http.Request) {
-		wsConn, err := upgrader.Upgrade(w, r, nil)
+	if *debug {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	pool := trcapi.NewPool(func() (*trcapi.Conn, func(), error) {
+		logger := log.WithField("path", *sock)
+
+		logger.Debug("Dialing Unix socket...")
+		unixConn, err := net.Dial("unix", *sock)
 		if err != nil {
-			http.Error(w, fmt.Sprintf("Failed to open WebSocket: %s", err), http.StatusBadRequest)
-			return
+			return nil, nil, errors.Wrapf(err, "failed to connect to TRC's unix socket")
 		}
-		defer wsConn.Close()
+		logger.Debug("Unix socket dial succeeded")
 
-		unixConn, err := net.Dial("unix", *sock)
+		logger.Debug("Initializing TRC protocol connection on Unix socket...")
+		trcConn, err := trcapi.Connect(trcapi.DefaultVersion, unixConn, unixConn)
 		if err != nil {
-			http.Error(w, fmt.Sprintf("Failed to connect to TRC's unix socket: %s", err), http.StatusInternalServerError)
-			return
+			return nil, nil, errors.Wrapf(err, "failed to establish connection to TRC")
 		}
-		defer unixConn.Close()
-
-		cl := api.NewClient(unixConn, unixConn)
-
-		go func() {
-			for {
-				select {
-				case <-r.Context().Done():
-					return
-				default:
-				}
-
-				var cmd api.Command
-				if err := wsConn.ReadJSON(&cmd); err != nil {
-					http.Error(w, fmt.Sprintf("Failed to read command: %s", err), http.StatusBadRequest)
-					continue
-				}
-
-				if err = cl.SendCommand(cmd); err != nil {
-					http.Error(w, fmt.Sprintf("Failed to send command to TRC: %s", err), http.StatusInternalServerError)
-					return
-				}
+		logger.Debug("TRC protocol connection initialized")
+
+		return trcConn, func() {
+			logger.WithError(err).Debug("Closing TRC connection...")
+			if err := trcConn.Close(); err != nil {
+				logger.WithError(err).Error("Failed to close TRC connection")
 			}
-		}()
 
-		for range time.Tick(time.Second) {
-			select {
-			case <-r.Context().Done():
-				return
-			default:
+			logger.WithError(err).Debug("Closing Unix socket...")
+			if err := unixConn.Close(); err != nil {
+				logger.WithError(err).Error("Failed to close Unix socket")
+			}
+		}, nil
+	})
+	defer pool.Close()
+
+	go func() {
+		var last time.Time
+		for {
+			d := time.Until(last.Add(retryInterval))
+			if d > 0 {
+				log.WithField("duration", d).Debug("Sleeping before retrying the connection...")
 			}
+			time.Sleep(d)
+
+			last = time.Now()
 
-			st, err := cl.State()
+			log.Debug("Retrieving a connection from pool...")
+			trcConn, err := pool.Conn()
```

Comment:
pool.Conn() returns one connection right? Then the server is very vulnerable. If for example an evil person sends a lot of connection requests then it will wait 5 seconds after each connection requests, so it will never be able to handle the connection request of a good person that sends a good connection request